drivers: sensor: lis2de12: fix read accel via spi

narangmayank opened the pull request:
Fixes: #83794 

Added a fix for reading acceleration data from LIS2DE12 sensor over SPI.

ubieda reviewed (changes requested):
Thanks for your contribution! Please squash both patches into a single one.

We shouldn't have one patch fixing something introduced by a previous commit in the same PR.

narangmayank committed:
drivers: sensor: lis2de12: fix read accel via spi

The lis2de12 sensor driver spi interface was calling spi read api.
This leads to a single byte operation on reading acceleration data
which is a multi byte operation. Fix it by adding a call to spi read
incr api instead. Added a length check to handle both single and multi
byte read properly.

Signed-off-by: Mayank Narang <narang.may77@gmail.com>

ubieda reviewed (dismissed):
Thanks for your contribution!

ubieda commented:
Adding backport labels as I think this is present on the last two releases. Feel free to overwrite if you disagree.

narangmayank commented:
@ubieda Thanks! I have a question regarding this patch, I'm using nrf connect sdk 2.8.0 which embed some commit of zephyr. Now since I made this patch here it will not gonna reflect in NCS v2.8.0 bundle so if any new developer joins tomorrow he needs to add this patch manually. Any chance I can override this in the app side with some application hook or something. Any suggestions?

ubieda commented:
> Now since I made this patch here it will not gonna reflect in NCS v2.8.0 bundle so if any new developer joins tomorrow he needs to add this patch manually. Any chance I can override this in the app side with some application hook or something. Any suggestions?

You can either keep this patch in your app (see `west patch`) or maintain Zephyr downstream with this patch applied. NCS will update Zephyr on subsequent releases to include your fix.

narangmayank commented:
Okay, I think west patch is coming in a future release of zephyr but anyways good to know about this. Its a great feature. I do not want to maintain whole zephyr code ideally instead I'm thinking to have a shell script for now which just dump  this patch into zephyr codebase while cloning the application.

avisconti commented on drivers/sensor/st/stmemsc/stmemsc_spi.c:
```diff
@@ -63,7 +63,10 @@ int stmemsc_spi_write(const struct spi_dt_spec *stmemsc,
+	if (len > 1) {
+		reg_addr |= STMEMSC_SPI_ADDR_AUTO_INCR;
+	}
+
```
I understand the rational behind it of course, but I'm curious to know if you did it to overcome some issue or just for the sake of doing the right thing. If so, wouldn't it be correct to change also the write_incr routine?

avisconti replied:
Would you min adding/testing the spi_read_incr one? I guess we may want to the the i2c part in another PR. Thanks! Then I'll give my +1

narangmayank replied:
I think you are asking for spi_write_incr one. I tested in the morning after your comment, it works fine. Let me add it.

avisconti replied:
> I think you are asking for spi_write_incr one. I tested in the morning after your comment, it works fine. Let me add it.

Yes, correct. Thanks!

narangmayank replied:
Added, Thank you :)

narangmayank committed:
drivers: sensor: lis2de12: add length check in spi write incr routine

Added a length check in the spi write incr routine to handle both
single and multi byte write operations properly.

Signed-off-by: Mayank Narang <narang.may77@gmail.com>
